pkg/commands/test/test.go

```diff
@@ -124,30 +139,61 @@ func processFile(ctx context.Context, fileName string, compiler *ast.Compiler) (
 	}
 
 	if err != nil {
-		return fmt.Errorf("Unable to open file %s: %s", fileName, err), nil
+		return nil, fmt.Errorf("Unable to open file %s: %s", fileName, err)
 	}
 
-	linebreak := detectLineBreak(data)
-	bits := bytes.Split(data, []byte(linebreak+"---"+linebreak))
+	return data, err
+}
 
+func processFile(fileName string) ([][]byte, parser.Parser, error) {
+	data, err := readDataFromFile(fileName)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to read data from file %s: %s", fileName, err)
+	}
+
+	linebreak := detectLineBreak(data)
+	elementList := bytes.Split(data, []byte(linebreak+"---"+linebreak))
 	parser := parser.GetParser(fileName)
+	return elementList, parser, nil
+}
 
+func checkPolicyOnElements(ctx context.Context, elementList [][]byte, unmarshaller parser.Parser, compiler *ast.Compiler) (error, error) {
 	var failuresList *multierror.Error
 	var warningsList *multierror.Error
-	for _, element := range bits {
-		var input interface{}
-		err = parser.Unmarshal([]byte(element), &input)
+	var failures, warnings error
+	var input []interface{}
```

Comment:
🤔 What about `var input map[string]interface{}` instead, where the keys are the file names? I imagine that it could be useful to refer to the different files in policies, instead of discarding that information... But I might be wrong, haven't fully caught up with the discussion yet.

Reply:
excellent idea. We've incorporated this feedback into a different PR to add this functionality.
thanks :)